src/lib/widget/_utilities/api.utilities.ts

```diff
@@ -1,254 +1,213 @@
+    const requestsUser = options?.user || true;
+    const { subscribe } = derived(
+      [journeyStore, oauthStore, userStore],
+      ([$journeyStore, $oauthStore, $userStore], set) => {
+        set({
+          journey: $journeyStore,
+          oauth: $oauthStore,
+          user: $userStore,
+        });
 
-      const oauthStoreUnsub = oauthStore.subscribe((response) => {
-        if (!requestsUser && response.successful) {
-          returnResponse &&
-            returnResponse({
-              journey,
-              oauth: response,
-            });
-          if (modal) {
-            modal.close({ reason: 'auto' });
+        if ($journeyStore.successful && $oauthStore.successful && $userStore.completed) {
+          modal && modal.close({ reason: 'auto' });
+        } else if ($journeyStore.successful && $oauthStore.successful) {
+          if (requestsUser && $userStore.loading === false && $userStore.completed === false) {
+            userStore.get();
+          } else if (!requestsUser) {
+            modal && modal.close({ reason: 'auto' });
           }
-        } else if (requestsUser && response.successful) {
-          oauth = response;
-          userStore?.get();
-        } else if (response.error) {
-          oauth = response;
-          returnError &&
-            returnError({
-              journey,
-              oauth: response,
-            });
-        }
-        /**
-         * Clean up unneeded subscription, but only when it's successful
-         * Leaving the subscription allows for the journey to be
-         * restarted internally.
-         */
-        if (response.successful) {
-          oauthStoreUnsub();
-        }
-      });
-      const userStoreUnsub = userStore.subscribe((response) => {
-        if (response.successful) {
-          returnResponse &&
-            returnResponse({
-              journey,
-              oauth,
-              user: response,
-            });
-          if (modal) {
-            modal.close({ reason: 'auto' });
+        } else if ($journeyStore.successful) {
+          if (requestsOauth && $oauthStore.loading === false && $oauthStore.completed === false) {
+            oauthStore.get();
+          } else if (!requestsOauth) {
+            modal && modal.close({ reason: 'auto' });
           }
-        } else if (response.error) {
-          returnError &&
-            returnError({
-              journey,
-              oauth,
-              user: response,
-            });
         }
-        /**
-         * Clean up unneeded subscription, but only when it's successful
-         * Leaving the subscription allows for the journey to be
-         * restarted internally.
-         */
-        if (response.successful) {
-          userStoreUnsub();
-        }
-      });
+      },
+    );
 
-      if (options?.resumeUrl) {
-        journeyStore.resume(options.resumeUrl);
+    function start(startOptions?: JourneyOptionsStart) {
```

Comment:
Here we're creating the `start` method and providing a returned Promise if the dev is only interested in the final event and its state. So, a dev can call `start` by itself and not assign the result of function call to anything, capturing the data in the `subscribe`. Or, the dev can use the `async-await` pattern and capture the returned result as a one-off.